### DISCUSSION TITLE
Error WIX0094: .wxs Files Not Loading

### DISCUSSION THREAD

**Discussion by burrito227** [wixtoolset/issues#7473] [Questions]
* Which version of WiX are you building with?

4.0.0

wix --version returns 4.0.0+8c757c0f

* Which version of Visual Studio are you building with (if any)?

2022

* Which version of .NET are you building with?

4.8

* Describe the problem and the steps to reproduce it.

I am a WiX newbie.

We recently converted our WiX v3 project to v4 using the heatwave tool. The build fails with the following error codes:

```
Error	WIX0094	The identifier 'WixComponentGroup:CMPGRP_MyProductWebServ' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.	MyProduct_Install	C:\Users\MyUser\Documents\MyProduct\branches\MyProduct_1.22\MyProduct_Install\Product.wxs	20	

Error	WIX0094	The identifier 'WixComponentGroup:CMPGRP_MyProductWinServ' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.	MyProduct_Install	C:\Users\MyUser\Documents\MyProduct\branches\MyProduct_1.22\MyProduct_Install\Product.wxs	26
```

The weird part is that the two files (MyProductWebServComp.wxs and MyProductWinServComp.wxs) are generated during the build. I can see the two files pop up in the solution explorer about 2-5 seconds before the build fails. If a second build is ran consecutively after the failed build, the second build succeeds since the old files are there.

We have the HeatDirectory defined in in our wixproj,

```
  <Target Name="BeforeBuild">
    <HeatDirectory Directory="$(OWinServiceBinariesDir)" PreprocessorVariable="var.OWinSPath" Transforms="WinServExcludes.xslt" OutputFile="MyProductWinServComp.wxs" ComponentGroupName="CMPGRP_MyProductWinServ" GenerateGuidsNow="true" DirectoryRefId="DIR_MyProductWINSERV" ToolPath="$(WixToolPath)" SuppressFragments="true" SuppressRegistry="true" SuppressRootDirectory="true" />
    <HeatDirectory Directory="$(OWebServiceBinariesDir)" PreprocessorVariable="var.OWebSPath" Transforms="WebServExcludes.xslt" OutputFile="MyProductWebServComp.wxs" ComponentGroupName="CMPGRP_MyProductWebServ" GenerateGuidsNow="true" DirectoryRefId="DIR_MyProductWEBSERV" ToolPath="$(WixToolPath)" SuppressFragments="true" SuppressRegistry="true" SuppressRootDirectory="true" />
  </Target>
```

And then an example in our Product.wxs file,

```
    <Feature Id="FEAT_Web" Title="MyProduct Web Service Installation" Level="1" ConfigurableDirectory="DIR_MyProductWEBSERV">
      <ComponentGroupRef Id="CMPGRP_MyProductWebServ" />
      <ComponentGroupRef Id="CMPGRP_WebServSetup" />
      <ComponentGroupRef Id="CMPGRP_WebConfig" />
    </Feature>
```

I saw a discussion where somebody was getting the same error, but sounded like different issue... I've tried adding with no change.

`<EnableProjectHarvesting>true</EnableProjectHarvesting>`

and

`<PackageReference Include="WixToolset.Heat" Version="4.*" />`

Does anybody have any idea what I am missing?

**Comment by BMurri** [comment]
Remove the `BeforeBuild` target after moving the two `HeatDirectory` elements out of it. Convert both into `HarvestDirectory` elements inside of an `ItemGroup`, changing the `Directory` attribute to `Include` and dropping the `OutputFile` and `ToolPath` attributes.

Note that it's not recommended to use `GenerateGuidsNow`, so I'd recommend removing that as well.
